src/pods/export/components/relation/components/database-relation-self-export.component.tsx

```diff
@@ -0,0 +1,90 @@
+import { Coords } from '@/core/model';
+import { RelationType, TABLE_CONST } from '@/core/providers';
+import { selfRelationcalculateOriginMinusFork } from '@/pods/canvas/components/relation/database-relation-self.business';
+import { isDrawLeftToRightLogic } from '@/pods/canvas/components/relation/relation.business';
+import React from 'react';
+import { ForkExportComponent } from './fork-export.component';
+
+interface Props {
+  relationType: RelationType;
+  startCoords: Coords;
+  endCoords: Coords;
+}
+
+export const DatabaseRelationSelfExportComponent: React.FC<Props> = props => {
+  const { relationType, startCoords, endCoords } = props;
+  const isDrawLeftToRight = isDrawLeftToRightLogic(
+    relationType,
+    startCoords,
+    endCoords
+  );
+
+  const originXMinusFork = selfRelationcalculateOriginMinusFork(startCoords);
+
+  const oneToOneRelationSelfPath = `
+  M ${startCoords.x} ${startCoords.y} 
+  H ${startCoords.x - TABLE_CONST.HORIZONTAL_LEFT_EXTENSION} 
+  V ${endCoords.y} 
+  H ${endCoords.x - TABLE_CONST.TABLE_WIDTH}
+  `;
+
+  const oneToManyRelationSelfPath = `
+  M ${startCoords.x} ${startCoords.y} 
+  H ${startCoords.x - TABLE_CONST.HORIZONTAL_LEFT_EXTENSION} 
+  V ${endCoords.y}
+  H ${originXMinusFork}
+  `;
+
+  const manyToOneRelationSelfPath = `
+  M ${originXMinusFork} ${startCoords.y} 
+  H ${startCoords.x - TABLE_CONST.HORIZONTAL_LEFT_EXTENSION} 
+  V ${endCoords.y}
+  H ${endCoords.x - TABLE_CONST.TABLE_WIDTH}
+  `;
+
+  const getRelationPathBasedOnType = (relationType: RelationType) => {
+    switch (relationType) {
+      case '1:M':
+        return oneToManyRelationSelfPath;
+      case 'M:1':
+        return manyToOneRelationSelfPath;
+      default:
+        return oneToOneRelationSelfPath;
+    }
+  };
+
+  return (
+    <svg>
+      {/* Glow filter if selected */}
+      <defs>
+        <filter id="table_glow">
+          <feGaussianBlur stdDeviation="3.5" result="coloredBlur" />
+          <feMerge>
+            <feMergeNode in="coloredBlur" />
+            <feMergeNode in="SourceGraphic" />
+          </feMerge>
+        </filter>
+      </defs>
+      <path
+        d={getRelationPathBasedOnType(relationType)}
+        fill="none"
+        stroke="#ecad5a"
```

Comment:
could we move to a constant becouse we can't use css here but we can use props or variables

Reply:
We are using this: 

```ts
export const exportStylesVariables = {
  PRIMARY_BACKGROUND: '#f1f5f9',
  SECONDARY_BACKGROUND: '#fefdfe',
  ACCENT_COLOR: '#e367f4',
  HIGHLIGHT_COLOR: '#e6eaf2',
  TEXT_COLOR: '#202020',
};
```

but not sure if for this case apply to create a new variable